### PR TITLE
Support oneTBB 2021

### LINKED ISSUE #1211: Adding support for oneTBB
Intels tbb library has released a new version under the name [oneTBB](https://github.com/oneapi-src/oneTBB), which is incompatible with the previous versions. As the new libs have not been designed to install alongside previous versions, all projects using tbb need updating simultaneously.

Currently OpenSubdiv does not compile with the first release of the oneTBB libs.

### PR DESCRIPTION
**Pull request by Logarithmus**
@jilliene

Closes #1211

I'm not sure my solution is right, I just did what [oneTBB's migration guide](https://docs.oneapi.io/versions/latest/onetbb/tbb_userguide/Migration_Guide/Task_Scheduler_Init.html) said.

### CODE REVIEW

**Comment by davidgyu** [comment]
Hi! Would you please submit a signed Contributor License Agreement as described here: [Contributing to OpenSubdiv](https://github.com/PixarAnimationStudios/OpenSubdiv/blob/release/documentation/contributing.rst#contributor-license-agreement) Thank you!

**Comment by Logarithmus** [comment]
@davidgyu would signing the PDF file with PGP and emailing it to you be sufficient?

**Comment by jilliene** [comment]
Filed as internal issue #OSD-359

**Comment by davidgyu** [comment]
A regular signature on the PDF emailed back to us would be best for our process. Thanks!

**Comment by Logarithmus** [comment]
@davidgyu I've just found a bit of free time and sent a signed PDF to you (opensubdiv-cla@pixar.com). It's signed both by hand & with GPG.

**Comment by davidgyu** [comment]
Thank you @Logarithmus !

File: opensubdiv/osd/tbbEvaluator.cpp [hunk 1]
```diff
@@ -215,10 +216,11 @@ TbbEvaluator::Synchronize(void *) {
...
 TbbEvaluator::SetNumThreads(int numThreads) {
-    if (numThreads == -1) {
-        tbb::task_scheduler_init init;
-    } else {
-        tbb::task_scheduler_init init(numThreads);
+    if (numThreads != -1) {
+		tbb::global_control tbb_global_control(
+			tbb::global_control::max_allowed_parallelism,
+			numThreads
+		);
```
Comment (anton-potapov) [inline opensubdiv/osd/tbbEvaluator.cpp:223]: Changes to TBB behavior made by  `tbb::global_control` only active wile the object is alive.  So having this object as stack variable makes the whole function `TbbEvaluator::SetNumThreads` to has no effects. 

From the [global_control::~global_control specification](https://spec.oneapi.io/versions/latest/elements/oneTBB/source/task_scheduler/scheduling_controls/global_control_cls.html) :
>Destructs a control variable object and ends it’s impact.

Comment (e4lam) [reply opensubdiv/osd/tbbEvaluator.cpp:223]: +1

Isn't the old code here also effectively a no-op too here? This entire `SetNumThreads()` method and references to it should be removed. A library like this has no business trying to control the number of threads used process wide. It should be a decision left to the library user instead.

**Comment by MoisMoshev** [comment]
Hey, I'm trying to build with your branch, but still unable to compile. Getting errors like these from msbuild:
```
"e:\bottleship\code\3p\OpenSubdiv\build\examples\dxViewer\dxViewer.vcxproj" (Rebuild target) (26) ->
(Link target) -> 
  LINK : fatal error LNK1104: cannot open file 'tbb12.lib'
```
and also
```
  E:\bottleship\code\3p\OpenSubdiv\examples\common\glPtexMipmapTexture.cpp(25,10): fatal error C1083: Cannot open include file: 'glLoader.h': No such file or directory
```

I'm trying with the following cmake options:
```
     -DTBB_INCLUDE_DIR='E:/bottleship/code/3p/oneapi-tbb-2021.5.0/include'
     -DTBB_tbb_LIBRARY='e:/bottleship/code/3p/oneapi-tbb-2021.5.0/lib/intel64/vc14/tbb.lib'
     -DTBB_tbb_debug_LIBRARY='e:/bottleship/code/3p/oneapi-tbb-2021.5.0/lib/intel64/vc14/tbb_debug.lib'
```
The libraries are there, but it's looking for tbb12, which I probably need to add to another path? 
Just wondering if my setup is the issue.

**Comment by svenstaro** [comment]
So what about merging this? Sadly this missed 3.5.0 but I'd appreciate it if this went in somewhat soon.

**Comment by davidgyu** [comment]
Thanks for your patience and apologies that this didn't make it into 3.5.0.
We realize this PR has been up for a long time. We'll add a note here when we know when to expect a release with this fix.

**Comment by e4lam** [comment]
I guess you could merge it but really, you might just as well delete the contents of the `SetNumThreads()` method as it's moving from a method that did nothing to a method that does nothing (as was commented >6 months ago).

**Comment by davidgyu** [comment]
Yes, I agree with your earlier comment regarding SetNumThreads()

**Comment by RBKreckel** [comment]
It would be extremely nice if this could be merged just to keep this package and depending packages building. In any case, it seems to me like the discussion about whether the function can be further simplified could be deferred until later.

**Comment by brad0** [comment]
As a OpenBSD package maintainer I came across this. It would be nice to have a proper fix upstream for TBB 2021 support.

**Comment by davidgyu** [comment]
As noted above, we'd like to address the presence of SetNumThreads() in the API along with the other low level fixes.
We have this on deck for our next release (3.6.x) where we'll be able to address API changes (the current 3.5.1 release is for minor bug fixes only). Thanks again for your patience.

**Comment by davidgyu** [comment]
Fixed by #1317 and #1319

**Comment by e4lam** [comment]
Thank you for the update!

**Pull request closed without merging**